tokenizers/src/tokenizer/encoding.rs

```diff
@@ -306,73 +343,95 @@ impl Encoding {
             return;
         }
 
+        assert!(stride < max_len);
+
         // Get the main overflowing part
-        let o_ids = self.ids.split_off(max_len);
-        let o_type_ids = self.type_ids.split_off(max_len);
-        let o_tokens = self.tokens.split_off(max_len);
-        let o_words = self.words.split_off(max_len);
-        let o_offsets = self.offsets.split_off(max_len);
-        let o_spe_toks = self.special_tokens_mask.split_off(max_len);
-        let o_attent = self.attention_mask.split_off(max_len);
-
-        // When truncating, we loose the `sequence_ranges` information.
+        let (o_ids, o_type_ids, o_tokens, o_words, o_offsets, o_spe_toks, o_attent) =
+            self.split_overflow(max_len, left);
+
+        // When truncating, we lose the `sequence_ranges` information.
         self.sequence_ranges.clear();
 
         // Now we need to separate the overflowing part into as many Encoding as needed
-        assert!(stride < max_len);
         let part_size = max_len - stride;
         let mut overflowing = vec![];
-        let mut part_id = 0;
+
+        let mut part_id = if left {
+            0
```

Comment:
This logic is confusing to me.
`part_id` is both incrementing and decrementing and we also use `left` in `get_current_part` within the rest of the loop.

Is there an option for remove all those `if` everywhere and to have 1 `if ` at most ?

```rust
let n = o_ids.len();
// This example is lacking the stride support but just a gist
let parts_offsets =if left {
 (0..n).step_by(part_size).filter_map(|start| {
    let stop = max(start + max_len, n);
    (start, stop)
});
} else{
  // This doesn't work but an example
  (0..n).step_by(-part_size).filter_map(|start| {
      let stop = max(start + max_len, n);
      (start, stop)
  }
}
```

And then remove `left` from the rest of the code.